darwin/DarwinProcessList.c

```diff
@@ -93,39 +95,40 @@ unsigned ProcessList_allocateCPULoadInfo(processor_cpu_load_info_t* p) {
    return cpu_count;
 }
 
-void ProcessList_getVMStats(vm_statistics_t p) {
+static void ProcessList_getVMStats(vm_statistics_t p) {
    mach_msg_type_number_t info_size = HOST_VM_INFO_COUNT;
 
    if (host_statistics(mach_host_self(), HOST_VM_INFO, (host_info_t)p, &info_size) != 0) {
       CRT_fatalError("Unable to retrieve VM statistics\n");
    }
 }
 
-struct kinfo_proc* ProcessList_getKInfoProcs(size_t* count) {
-   int mib[4] = { CTL_KERN, KERN_PROC, KERN_PROC_ALL, 0 };
+static struct kinfo_proc* ProcessList_getKInfoProcs(size_t* count) {
+   const int mib[4] = { CTL_KERN, KERN_PROC, KERN_PROC_ALL, 0 };
    struct kinfo_proc* processes = NULL;
 
-   /* Note the two calls to sysctl(). One to get length and one to get the
-    * data. This -does- mean that the second call could end up with a missing
-    * process entry or two.
-    */
-   *count = 0;
-   if (sysctl(mib, 4, NULL, count, NULL, 0) < 0) {
-      CRT_fatalError("Unable to get size of kproc_infos");
-   }
+   int retries = 3;
 
-   processes = xMalloc(*count);
-   if (processes == NULL) {
-      CRT_fatalError("Out of memory for kproc_infos");
-   }
+   while (retries > 0) {
+      retries--;
```

Comment:
Use for loop? OR at least inline the decrement?

Reply:
switched to for loop